keyboards/keycapsss/kimiko/rev2/keymaps/default/keymap.c

```diff
@@ -0,0 +1,345 @@
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                                           KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______,
+    KC_F1,  KC_F2,    KC_F3,   KC_F4,   KC_F5,   KC_F6,                                          XXXXXXX, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, XXXXXXX,
+    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,   _______, _______,  _______, _______,  KC_PLUS, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
+                               _______, _______, _______,  _______, _______,  _______, _______,  _______, _______, _______
+),
+/* ADJUST (Press LOWER and RAISE together)
+ * ,--------------------------------------------.                                   ,----------------------------------------------.
+ * | QK_BOOT |      |      |      |     |       |                                   |      |      |      |      |      |           |
+ * |---------+------+------+------+------+------|                                   |------+------+------+------+------+-----------|
+ * | RGB ON  | HUE+ | SAT+ | VAL+ |      |      |                                   | PREV | PLAY | NEXT |      |      |           |
+ * |---------+------+------+------+------+------|                                   |------+------+------+------+------+-----------|
+ * | MODE    | HUE- | SAT- | VAL- |      |      |---------------.   ,---------------| VOL+ | MUTE | VOL- |      |      |           |
+ * |---------+------+------+------+------+------|      |        |   |       |       |------+------+------+------+------+-----------|
+ * |         |      |      |      |      |      |------|--------|   |-------|-------|      |      |      |      |       |           |
+ * `--------------------------------------------|               /    \              |----------------------------------------------'
+ *                         | LCTRL| LGUI |LALT  |LOWER | Space /      \Enter| RAISE |BackSP| RGUI | RALT |
+ *                         `----------------------------------'        '---------------------------------'
+ */
+[_ADJUST] = LAYOUT(
+    QK_BOOT,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, XXXXXXX, XXXXXXX,                                     KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX, XXXXXXX,
+    RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, XXXXXXX, XXXXXXX,                                     KC_VOLU, KC_MUTE, KC_VOLD, XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+                               _______, _______, _______, _______, _______, _______, _______,  _______, _______, _______
+)
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    state = update_tri_layer_state(state, _RAISE, _LOWER, _ADJUST);
+    return state;
+}
+
+
+#ifdef OLED_ENABLE
```

Comment:
This oled code should be moved to the keyboard level, (in this case, maybe rev2.c, if the rev1 doesn't have an oled).  This way, each keymap doesn't need to have code for the oled code but will have a display (this is especially useful for the configurator)

Reply:
Thank you for the suggestion.
The Rev1 has also a oled display. So i could move it to kimiko.c?

What is if a user wants to customise the oled display? Can it be overwritten in the keymap?

Reply:
Then yes, it should be moved to the kimiko.c file.

And yes, specifically you should be using: 
```c
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return;
    }
    // keyboard code

    return true;
}
```

This allows you to specify the keyboard code. oled_task_user returns true by default, and returning false allows you to override the keyboard level code.

Reply:
Thank you for pointing in the right direction.
I checked other keyboard codes like 'crkbd' and they return `false` at the end.
Should i return `true` or `false` in the `kimiko.c`?
https://github.com/qmk/qmk_firmware/blob/f07490bc092e365ba03dc685b3fc30ad0bf0b752/keyboards/crkbd/crkbd.c#L144-L155

Reply:
Honestly, it doesn't matter.  It's not used for anything, currently, so there is no difference.